### PR TITLE
case insensitive parsing of configuration file names

### PR DESCRIPTION
**Pull request by nickhand**
* implements case insensitive parsing of names when comparing names in the Schema and configuration file 
* adds some more documentation

Is that what you had in mind @rainwoodman ?

### CODE REVIEW

**Comment by rainwoodman** [comment]
Looks about right, but which name is from the file and which name is for the schema?

File: nbodykit/utils/config.py [hunk 1]
```diff
@@ -65,31 +65,32 @@ def fill_namespace(ns, arg, config, missing):
         a list to add arguments that are missing, i.e., required and not
         present in the input configuration files
     """
-    # the name of the parameter
-    name = arg.name.split('.')[-1]
+    # the name of the parameter (as taken from the schema)
+    schema_name = arg.name.split('.')[-1]
```
Comment (nickhand) [inline nbodykit/utils/config.py]: this is the name of the argument taken from the schema

File: nbodykit/utils/config.py [hunk 2]
```diff
@@ -65,31 +65,32 @@ def fill_namespace(ns, arg, config, missing):
...
     
     # no subfields
     if not len(arg.subfields): 
     
-        if config is not None and name.lower() in [k.lower() for k in config]:
-            value = config.pop(name)
+        # check if the schema argument is present in configuration file
+        if config is not None and schema_name.lower() in [k.lower() for k in config]:
```
Comment (nickhand) [inline nbodykit/utils/config.py]: and the keys of the ``config`` dict here are the names of the parameters from the configuration file

File: nbodykit/utils/config.py [hunk 3]
```diff
@@ -30,31 +38,59 @@ def construct_mapping(loader, node):
...
     
-    if not len(arg.subfields):  
-        if config is not None and name in config:
-            value = config.pop(name)
+    # no subfields
+    if not len(arg.subfields): 
+    
+        # check if the schema argument is present in configuration file
+        if config is not None and schema_name.lower() in [k.lower() for k in config]:
+            value = config.pop(schema_name)
```
Comment (rainwoodman) [inline nbodykit/utils/config.py]: Will this fail if the cases don't match.

Comment (nickhand) [inline nbodykit/utils/config.py]: ah yes, yes it will

**Comment by nickhand** [comment]
okay to close #214 ?

**Pull request merged**